client/client.go

```diff
@@ -347,25 +351,50 @@ func (r *NotaryRepository) Publish() error {
 		return err
 	}
 
+	// these are the tuf files we will need to update, serialized as JSON before
+	// we send anything to remote
+	update := make(map[string][]byte)
+
 	// check if our root file is nearing expiry. Resign if it is.
 	if nearExpiry(r.tufRepo.Root) || r.tufRepo.Root.Dirty {
-		if err != nil {
-			return err
-		}
 		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"))
+		updateRoot = true
+	}
+
+	if updateRoot {
```

Comment:
Because you've improved the code layout I think you can just delete this check and move the inner block inside the preceding `if`, deleting `updateRoot = true` at the same time.